### ISSUE TITLE
BufferAllocator.h:60:29: error: implicit instantiation of undefined template 'std::__1::array<bool, 16>'

### ISSUE THREAD

**Issue by perfect7gentleman** [whoozle/android-file-transfer-linux#243]
Error:
```
ninja: job failed: /usr/lib/ccache/bin/clang++ -DHAVE_LIBMAGIC -DHAVE_TAGLIB=1 -DMTPZ_ENABLED -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE=1 -Dmtp_ng_EXPORTS -I/tmp/portage/sys-fs/android-file-transfer-linux-9999/work/android-file-transfer-linux-9999 -I/tmp/portage/sys-fs/android-file-transfer-linux-9999/work/android-file-transfer-linux-9999/mtp/backend/posix -I/tmp/portage/sys-fs/android-file-transfer-linux-9999/work/android-file-transfer-linux-9999/mtp/backend/linux -isystem /usr/include/taglib  -march=native -mtune=native -O3 -pipe -flto=thin -fomit-frame-pointer -fno-plt -fno-stack-protector -fno-unwind-tables -stdlib=libc++ -fPIC   -Wall -pthread -std=gnu++11 -MD -MT CMakeFiles/mtp-ng.dir/mtp/backend/linux/usb/Device.cpp.o -MF CMakeFiles/mtp-ng.dir/mtp/backend/linux/usb/Device.cpp.o.d -o CMakeFiles/mtp-ng.dir/mtp/backend/linux/usb/Device.cpp.o -c /tmp/portage/sys-fs/android-file-transfer-linux-9999/work/android-file-transfer-linux-9999/mtp/backend/linux/usb/Device.cpp
In file included from /tmp/portage/sys-fs/android-file-transfer-linux-9999/work/android-file-transfer-linux-9999/mtp/backend/linux/usb/Device.cpp:20:
/tmp/portage/sys-fs/android-file-transfer-linux-9999/work/android-file-transfer-linux-9999/mtp/backend/linux/usb/BufferAllocator.h:60:29: error: implicit instantiation of undefined template 'std::__1::array<bool, 16>'
                std::array<bool, Buffers> _bufferAllocated;
                                          ^
/usr/include/c++/v1/__tuple:219:64: note: template is declared here
template <class _Tp, size_t _Size> struct _LIBCPP_TEMPLATE_VIS array;
                                                               ^
In file included from /tmp/portage/sys-fs/android-file-transfer-linux-9999/work/android-file-transfer-linux-9999/mtp/backend/linux/usb/Device.cpp:20:
In file included from /tmp/portage/sys-fs/android-file-transfer-linux-9999/work/android-file-transfer-linux-9999/mtp/backend/linux/usb/BufferAllocator.h:23:
In file included from /tmp/portage/sys-fs/android-file-transfer-linux-9999/work/android-file-transfer-linux-9999/mtp/ByteArray.h:23:
In file included from /tmp/portage/sys-fs/android-file-transfer-linux-9999/work/android-file-transfer-linux-9999/mtp/types.h:27:
/usr/include/c++/v1/memory:3861:24: error: assigning to 'std::__1::__shared_weak_count *' from incompatible type 'std::__1::__shared_ptr_emplace<mtp::usb::BufferAllocator, std::__1::allocator<mtp::usb::BufferAllocator> > *'
        __r.__cntrl_ = __cntrl;
                       ^~~~~~~
/usr/include/c++/v1/memory:4403:29: note: in instantiation of function template specialization 'std::__1::shared_ptr<mtp::usb::BufferAllocator>::__create_with_control_block<mtp::usb::BufferAllocator, std::__1::__shared_ptr_emplace<mtp::usb::BufferAllocator, std::__1::allocator<mtp::usb::BufferAllocator> > >' requested here
    return shared_ptr<_Tp>::__create_with_control_block(__ptr, __hold2.release());
                            ^
/tmp/portage/sys-fs/android-file-transfer-linux-9999/work/android-file-transfer-linux-9999/mtp/backend/linux/usb/Device.cpp:83:27: note: in instantiation of function template specialization 'std::__1::make_shared<mtp::usb::BufferAllocator, int>' requested here
                _bufferAllocator = std::make_shared<BufferAllocator>(mmap? fd: -1);
                                        ^
2 errors generated.
```

Solution:
add
```
#include <array>
```
to android-file-transfer-linux-9999/mtp/backend/linux/usb/BufferAllocator.h

**Comment by whoozle** [comment]
try pulling, should be fixed in b6c614d2fa28f65b87fbb5747bd358a95359990a

**Comment by perfect7gentleman** [comment]
fixed.
